palace/linalg/pc.hpp

```diff
@@ -1,8 +1,8 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-#ifndef PALACE_KSP_PRECONDITIONER_HPP
-#define PALACE_KSP_PRECONDITIONER_HPP
+#ifndef PALACE_LINALG_KSP_PRECONDITIONER_HPP
+#define PALACE_LINALG_KSP_PRECONDITIONER_HPP
```

Comment:
The KSP doesn't match the file name, and if I understand the acronym correctly, this reads as "Krylov Subspace Preconditioner Preconditioner" which is a bit ATM machine.

Reply:
As above.

Reply:
As above, the include guard matches the `KspPreconditioner` class name. KSP is a PETSc acronym for Krylov subspace solvers, the "P" is not "Preconditioner" (might be better described as PKS solver, preconditioned Krylov subspace solver, or maybe the SP is for SubsPace, either way not our problem here).